neurodamus/metype.py

```diff
@@ -35,6 +36,63 @@ def connect2target(self, target_pp=None):
         """Connects empty cell to target"""
         return Nd.NetCon(self._cellref, target_pp)
 
+    @staticmethod
+    def get_sec(cell, section_id):
+        """Return the section corresponding to the global section_id
+        Note: supports python's negative indices.
+        """
+        # Build a list of (section_list, length) tuples for available section types
+        sec_lists = [getattr(cell, stype) for stype in BaseCell._sections if hasattr(cell, stype)]
+
+        # Compute total length
+        total_len = sum(len(lst) for lst in sec_lists)
+
+        # Handle negative index
+        if section_id < 0:
+            section_id += total_len
+
+        if not (0 <= section_id < total_len):
+            raise IndexError(
```

Comment:
`IndexError` is for subscripts, but we aren't in a `__getitem__` method, so it's a bit of a misnomer.

Reply:
mmm I do not agree for a few reasons:

- It does not need to be in `__getitem__` to be valid.
- `IndexError` is the standard exception for out-of-bounds access in Python, and it’s used in many APIs beyond just subscripting (e.g. list.pop(i), next(it) with __getitem__-like behavior, etc.).
- The semantics is very close to subscript access: I am resolving a flat `section_id` into the correct item: conceptually, this is indexing.

Reply:
Hmm, I'm not sure it's that commonly used except in the explicit container case in python core.  A couple for empty containers, list.pop, MutableSequence.insert.  What do you mean by `next(it)`?

Reply:
cant next(it) raise an indexError? I thought yes

Reply:
Sorry for being grump, but I think consistency is important.

What about making a `OutOfRangeSectionIDError` or something?

Reply:
ah not even an existing one. Ok, I will think about it